### ISSUE TITLE
UsdUVTexture alpha seems to come from red channel

### ISSUE THREAD

**Issue by ablev-dwa** [PixarAnimationStudios/USD#657]
### Description of Issue

When creating a UsdPreviewSurface where we want to leverage opacity, it seems that the alpha channel output on a UsdUVTexture (outputs:a) is being driven by the red channel in the image. I've attached a simple repro (preview_alpha.zip) that demonstrates the issue. Where the red channel is 1, the shading result is opaque, but as the red channel goes to zero, so does the opacity.

```
#usda 1.0
(
    defaultPrim = "card"
)

def Xform "card"
{
    def Scope "proxy"
    {
        uniform token purpose = "proxy"
        
        def Mesh "card"
        {
            int[] faceVertexCounts = [4]
            int[] faceVertexIndices = [0, 1, 2, 3]
            point3f[] points = [(-1, -1, 0), (-1, 1, 0), (1, 1, 0), (1, -1, 0)] (
                interpolation = "vertex"
            )
            texCoord2f[] primvars:uv = [(0, 0), (0, 1), (1, 1), (1, 0)] (
                interpolation = "faceVarying"
            )
            uniform bool doubleSided = true
            rel material:binding:preview = </card/Looks/preview>
        }
    }
    
    def Scope "Looks"
    {
        def Material "preview"
        {

            token outputs:surface.connect = </card/Looks/preview/preview_surface.outputs:surface>

            def Shader "preview_surface"
            {
                uniform token info:id = "UsdPreviewSurface"
                token outputs:surface
                color3f inputs:diffuseColor = (0, 0, 0)

                int inputs:useSpecularWorkflow = 1
                color3f  inputs:specularColor           = (0, 0, 0)
                color3f  inputs:emissiveColor.connect   = </card/Looks/preview/preview_map.outputs:rgb>
                float    inputs:displacement            = 0.0
                float    inputs:opacity.connect         = </card/Looks/preview/preview_map.outputs:a>
                float    inputs:roughness               = 1.00
                float    inputs:metallic                = 0.00
                float    inputs:clearcoat               = 0.00
                float    inputs:clearcoatRoughness      = 1.00
                float    inputs:occlusion               = 0.00
                normal3f inputs:normal                  = (0, 0, 1)
                float    inputs:ior                     = 0.00
            }

            def Shader "preview_map"
            {
                uniform token info:id = "UsdUVTexture"
                asset inputs:file = @./atlas.png@
                float4 inputs:fallback = (0, 1, 0, 0)
                float2 inputs:st.connect = </card/Looks/preview/read_uv.outputs:result>
                token inputs:wrapS = "black"
                token inputs:wrapT = "clamp"
                float3 outputs:rgb
                float outputs:a
            }
            def Shader "read_uv"
            {
                uniform token info:id = "UsdPrimvarReader_float2"
                token inputs:varname = "uv"
                float2 outputs:result
            }
        }
    }
}
```

### Steps to Reproduce
1. usdview card.usd

### System Information (OS, Hardware)
RHEL6.7

### Package Versions
18.9, 18.11

### Build Flags
 python build_scripts/build_usd.py /usr/pic1/usdtest --tests

[preview_alpha.zip](https://github.com/PixarAnimationStudios/USD/files/2474940/preview_alpha.zip)

**Comment by jtran56** [comment]
Filed as internal issue #USD-4828.

**Comment by elrond79** [comment]
I encountered this too... apparently HdSt is actually completely ignoring the name of the specific parameter - ie, if you did:

```
color3f  inputs:emissiveColor.connect   = </card/Looks/preview/preview_map.outputs:doesnt_exist>
float    inputs:opacity.connect         = </card/Looks/preview/preview_map.outputs:blahblah>
```

...it still renders without complaint, though you just get the rgb / r in each case, respectively.  (Also - UsdUVTexture's definition in shaderDefs.usda currently doesn't even define an "outputs:a" - just r/g/b/rgb/rgba.  So connecting to outputs:a currently is really the same as connecting to ouputs:doesnt_exist...)

I guess codeGen needs to be updated so the constructed HdGet_* actually pay attention to what parameter it's connected to.

**Comment by mistafunk** [comment]
Just encountered this in 19.07, is there already a patch or something i could apply locally?

**Comment by mistafunk** [comment]
This is still happening in USDView 20.02 and it seems Houdini (looked at 18.0.416) is also affected.

**Comment by spiffmon** [comment]
Good news, @mistafunk : we are working on the underlying cause of this problem now, and should have it addressed for 20.08

**Comment by mistafunk** [comment]
@spiffmon great news, thanks! Do you expect this work to result in a patch which we could put on a custom build based on 20.02 before mid-may? (our product releases early june and then only again in october or so...)

**Comment by spiffmon** [comment]
@mistafunk,
   You're welcome to try patching the change when it comes out, though this area of the code is under very active development, so it may or may not be straightforward.

Cheers,
--spiff

**Comment by mistafunk** [comment]
Great to have the fix already for 20.05, not 20.08, thanks!
